x/reward/abci_test.go

```diff
@@ -0,0 +1,45 @@
+package reward_test
+
+import (
+	"testing"
+	"time"
+
+	simapp "github.com/provenance-io/provenance/app"
+	"github.com/provenance-io/provenance/x/reward"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+func TestEndBlockWithNoActiveRewards(t *testing.T) {
+	var app *simapp.App
+	var ctx sdk.Context
+
+	now := time.Now()
+
+	app = simapp.Setup(false)
+	ctx = app.BaseApp.NewContext(false, tmproto.Header{})
+	ctx = ctx.WithBlockHeight(1).WithBlockTime(now)
+
+	ctx = ctx.WithBlockHeight(2)
+	require.NotPanics(t, func() {
+		reward.EndBlocker(ctx, app.RewardKeeper)
+	})
+}
+
+func TestBeginBlockWithNoActiveRewards(t *testing.T) {
+	var app *simapp.App
+	var ctx sdk.Context
+
+	now := time.Now()
+
+	app = simapp.Setup(false)
+	ctx = app.BaseApp.NewContext(false, tmproto.Header{})
+	ctx = ctx.WithBlockHeight(1).WithBlockTime(now)
+
+	ctx = ctx.WithBlockHeight(2)
+	require.NotPanics(t, func() {
+		reward.BeginBlocker(ctx, app.RewardKeeper)
+	})
+}
```

Comment:
i think the Begin blocker and end blocker is checked in keeper_test.go quite extensively..for e.g ```	suite.app.RewardKeeper.StartRewardProgram(suite.ctx, &rewardProgram)
	suite.app.RewardKeeper.SetRewardProgram(suite.ctx, rewardProgram)

	// We want to set the events here
	validators := getTestValidators(6, 6)
	delegates := suite.createDelegateEvents(validators[0].OperatorAddress, "1000000000nhash", "cosmos15ky9du8a2wlstz6fpx3p4mqpjyrm5cgxwpuzvh", "50000000000000.000000000000000000")
	delegates = delegates.AppendEvents(suite.createDelegateEvents(validators[0].OperatorAddress, "1000000000nhash", "cosmos15ky9du8a2wlstz6fpx3p4mqpjyrm5cgxwpuzvh", "50000000000000.000000000000000000"))
	SetupEventHistory(suite, delegates)
	reward.EndBlocker(suite.ctx, suite.app.RewardKeeper)```
Additionally the msg_servicerouter_test also do this because they exercise this via ```app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: height, Time: time.Now().UTC()}})``` &&& ```app.EndBlock(abci.RequestEndBlock{Height: height})
		app.Commit()``` ..so i would defer to those tests imo..i would almost vote to delete `abci_test.go` because it give the impression that those were not tested more elaborately but 🤷‍♂️